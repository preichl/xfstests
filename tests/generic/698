#! /bin/bash
# SPDX-License-Identifier: GPL-2.0
# Copyright (c) 2022  Red Hat Inc. All Rights Reserved.
#
# FS QA Test 698
#
# Verify that i_blocks for truncated files larger than 4 GiB have correct
# values.
#
# This test verifies the problem fixed in kernel with commit
# 92fba084b79e exfat: fix i_blocks for files truncated over 4 GiB
#
. ./common/preamble
. ./common/filter

_begin_fstest auto

# Override the default cleanup function.
_cleanup()
{
	cd /
	rm -r -f $tmp.* $junk_dir
}

_supported_fs generic
_fixed_by_kernel_commit 92fba084b79e \
	"exfat: fix i_blocks for files truncated over 4 GiB"

_require_test
_require_fs_space $TEST_DIR $((5 * 1024 * 1024)) #kB

junk_dir=$TEST_DIR/$seq
junk_file=$junk_dir/junk
mkdir -p $junk_dir

$XFS_IO_PROG -f -c "pwrite -W 0 5G" $junk_file > /dev/null

truncate -s $((4 * 1024 * 1024 * 1024)) $junk_file

block_size=`stat -c '%B' $junk_file`
iblocks_after_truncate=`stat -c '%b' $junk_file`
iblocks_expected=$((4 * 1024 * 1024 * 1024 / $block_size))

_within_tolerance "Number of allocated blocks after truncate" $iblocks_after_truncate $iblocks_expected 1% -v

status=0

exit
