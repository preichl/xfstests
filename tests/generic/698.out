QA output created by 698
Number of allocated blocks after truncate is in range
